### PR TITLE
Added overridable auto-login to GlobusApp

### PR DESCRIPTION
**Pull request by derek-globus**
Added overridable auto-login to GlobusApp

----
📚 Documentation preview 📚: https://globus-sdk-python--994.org.readthedocs.build/en/994/

### CODE REVIEW

File: src/globus_sdk/experimental/globus_app/_validating_token_storage.py [hunk 1]
```diff
@@ -108,31 +114,31 @@ def store_token_data_by_resource_server(
...
-    def get_token_data(self, resource_server: str) -> TokenData | None:
+    def get_token_data(self, resource_server: str) -> TokenData:
         """
         :param resource_server: A resource server with cached token data.
-        :returns: The token data for the given resource server, or None if no token data
-            is present in the attached storage adapter.
-        :raises: :exc:`TokenValidationError` if the token has expired or does not meet
-            the attached scope requirements.
+        :returns: The token data for the given resource server.
+        :raises: :exc:`TokenValidationError` to signal a problem with stored token data.
```
Comment (kurtmckee) [inline src/globus_sdk/experimental/globus_app/_validating_token_storage.py:134]:
```suggestion
        :raises: :exc:`MissingTokensError` to signal a problem with stored token data.
```

File: src/globus_sdk/experimental/globus_app/authorizer_factory.py [hunk 2]
```diff
@@ -141,9 +153,10 @@ def _make_authorizer(self, resource_server: str) -> RefreshTokenAuthorizer:
         :param resource_server: The resource server the authorizer will produce
             authentication for
         """
-        token_data = self._get_token_data_or_error(resource_server)
+        token_data = self.token_storage.get_token_data(resource_server)
         if token_data.refresh_token is None:
-            raise MissingTokensError(f"No refresh_token for {resource_server}")
+            msg = f"No refresh_token for {resource_server}"
+            raise MissingTokensError(msg, resource_server=resource_server)
```
Comment (kurtmckee) [inline src/globus_sdk/experimental/globus_app/authorizer_factory.py:159]: The docstring isn't implicated in this change so I can't highlight it, but for consistency with other changes, could the narrative "Raises blah blah blah" convert to autodoc-compatible markup, like `:raises: :exc:...`?

**Review by kurtmckee — CHANGES_REQUESTED** [review]

File: src/globus_sdk/experimental/globus_app/_validating_token_storage.py (hunk 1 above)
Comment (derek-globus) [reply src/globus_sdk/experimental/globus_app/_validating_token_storage.py:134]: `TokenValidationError` is the base subclass of errors raised by this method (of which `MissingTokensError` is one) and multiple different `TokenValidationError` types can be raised from this method.

The basic error heirarchy structure is
  * "TokenValidationError" - category of errors
     * `MissingTokensError` - No token exists for resource server.
     * `ExpiredTokenError` - Token exists but is expired.
     * `UnmetScopeRequirementsError` - Token exists but root or dependent scope requirements are unmet.
     * "IdentityMismatchError" - category of errors
        * `MissingIdentityError` - Token response does not include identity info.
        * `IdentityMismatchError` - Token response includes identity info but it differs from the stored identity.

With the latest rewrite, `get_token_error` does only raise 2 of these error types: [`MissingTokensError`, `UnmetScopeRequirementsError`] (`ExpiredTokenError` was pulled based on a discussion Aaron & I had about how RefreshAuthorizerFactory should function in the case of an expired access token) so I suppose I could scope the docstring down to just those two?

File: src/globus_sdk/experimental/globus_app/authorizer_factory.py (hunk 2 above)
Comment (derek-globus) [reply src/globus_sdk/experimental/globus_app/authorizer_factory.py:159]: Sure, updated in new rev

File: src/globus_sdk/experimental/globus_app/_validating_token_storage.py (hunk 1 above)
Comment (derek-globus) [reply src/globus_sdk/experimental/globus_app/_validating_token_storage.py:134]: Updated this & other docstrings to define more concretely which errors they raise.

**Review by kurtmckee — COMMENTED** [review]
GitHub is indicating that all 10 files changed, perhaps due to a rebase or rewrite. I'm not able to track what changed since I left comments.

**Review by kurtmckee — APPROVED** [review]

**Pull request merged**